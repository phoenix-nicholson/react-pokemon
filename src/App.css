body {
  font-family: 'VT323', monospace;
  font-size: 1.5rem;
  background-color: plum;
}
img {
  width: 160px;
  height: auto;
}
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.loader {
  transform: translateZ(1px);
}
.loader:after {
  margin: 20px;
  content: '$';
  display: inline-block;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  text-align: center;
  line-height: 40px;
  font-size: 32px;
  font-weight: bold;
  background: #ffd700;
  color: #daa520;
  border: 4px double;
  box-sizing: border-box;
  box-shadow: 2px 2px 2px 1px rgba(0, 0, 0, 0.1);
  animation: coin-flip 4s cubic-bezier(0, 0.2, 0.8, 1) infinite;
}
@keyframes coin-flip {
  0%,
  100% {
    animation-timing-function: cubic-bezier(0.5, 0, 1, 0.5);
  }
  0% {
    transform: rotateY(0deg);
  }
  50% {
    transform: rotateY(1800deg);
    animation-timing-function: cubic-bezier(0, 0.5, 0.5, 1);
  }
  100% {
    transform: rotateY(3600deg);
  }
}
