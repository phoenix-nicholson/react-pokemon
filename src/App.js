import './App.css';
import { useEffect, useState } from 'react';
import { getPokemon, getType } from './services/poke';
import Control from './components/controls/Controls';
import PokeList from './components/PokeList/PokeList';

function App() {
  const [pokemon, setPokemon] = useState([]);
  const [query, setQuery] = useState('');
  const [loading, setLoading] = useState(true);
  const [order, setOrder] = useState('asc');
  const [page, setPage] = useState(1);
  const [type, setType] = useState([]);
  const [selectedType, setSelectedType] = useState('all');

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPokemon(query, order, page, selectedType);
      setPokemon([...data.results]);
      setTimeout(() => {
        setLoading(false);
      }, 1500);
    };
    if (loading) {
      fetchData();
    }
  }, [loading, query, order, page, selectedType]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getType();
      setType(data);
    };
    fetchData();
  }, []);

  return (
    <div className="App">
      <h1>Pokedex</h1>
      <Control
        query={query}
        setQuery={setQuery}
        setLoading={setLoading}
        order={order}
        setOrder={setOrder}
        type={type}
        setSelectedType={setSelectedType}
        selectedType={selectedType}
      />
      {loading && <span className="loader"></span>}
      {!loading && <PokeList pokemon={pokemon} setLoading={setLoading} setPage={setPage} />}
    </div>
  );
}

export default App;
