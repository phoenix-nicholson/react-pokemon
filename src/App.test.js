import { render } from '@testing-library/react';
import App from './App';

test('renders pokes correctly', () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});
