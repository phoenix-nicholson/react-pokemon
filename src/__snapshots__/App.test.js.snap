// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pokes correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          Pokedex
        </h1>
        <div>
          <input
            placeholder="Search Pokettomonsuta"
            type="text"
            value=""
          />
          <button>
            Search
          </button>
          <select>
            <option
              value="asc"
            >
              Ascending
            </option>
            <option
              value="desc"
            >
              Descending
            </option>
          </select>
          <select>
            <option
              value="all"
            >
              All
            </option>
          </select>
        </div>
        <span
          class="loader"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        Pokedex
      </h1>
      <div>
        <input
          placeholder="Search Pokettomonsuta"
          type="text"
          value=""
        />
        <button>
          Search
        </button>
        <select>
          <option
            value="asc"
          >
            Ascending
          </option>
          <option
            value="desc"
          >
            Descending
          </option>
        </select>
        <select>
          <option
            value="all"
          >
            All
          </option>
        </select>
      </div>
      <span
        class="loader"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
